### PR TITLE
[#732] Create status-go API to return a current revision

### LINKED ISSUE #732: Create status-go API to return a current revision
# Problem

As a developer, I want to see the revision of status-go in Status. Currently, the UI part has no way to retrieve it.

# Implementation

See `statusd -version` command implementation

# Acceptance Criteria

* The API is added that returns the current revision as string.
* The API is exported when building via xgo

### PR DESCRIPTION
**Pull request by ovx**
Change `BUILD_FLAGS` in Makefile and update `github.com/status-im/status-go/geth/params.VersionMeta` value with the current Git SHA.

Closes #732

### CODE REVIEW

File: geth/api/api.go [hunk 1]
```diff
@@ -248,3 +248,8 @@ func (api *StatusAPI) AppStateChange(state string) {
 	}
 	api.b.AppStateChange(appState)
 }
+
+// StatusGoVersion returns build version
+func (api *StatusAPI) StatusGoVersion() string {
```
Comment (adambabik) [inline geth/api/api.go]: I think it can be named `Version()`. There are no other components than `StatusGo`.

File: lib/library.go [hunk 2]
```diff
@@ -461,3 +461,10 @@ func ConnectionChange(typ *C.char, expensive C.int) {
 func AppStateChange(state *C.char) {
 	statusAPI.AppStateChange(C.GoString(state))
 }
+
+// StatusGoVersion returns build version
+//export StatusGoVersion
+func StatusGoVersion() *C.char {
```
Comment (adambabik) [inline lib/library.go]: This also can be named `Version`.

File: t/e2e/api/api_test.go [hunk 3]
```diff
@@ -274,3 +274,20 @@ func (s *APITestSuite) TestNodeStartCrash() {
 	// cleanup
 	s.NoError(s.api.StopNode())
 }
+
+func (s *APITestSuite) TestStatusGoVersion() {
+	var (
```
Comment (adambabik) [inline t/e2e/api/api_test.go]: `var` is not needed, `require := s.Require()` will work just fine.

Also, I don't think you need to start the node here. 

```go
version := s.api.StatusGoVersion()
s.Equal(params.Version, version)
```

should be enough.

**Review by adambabik — CHANGES_REQUESTED** [review]

**Comment by dshulyak** [comment]
are we certain that we need a binding for that and not rpc call `status_version` such as https://github.com/ethereum/wiki/wiki/JSON-RPC#web3_clientversion ?

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/ethereum/wiki">ethereum/wiki</a></strong></div><div>wiki - The Ethereum Wiki - </div></blockquote>

**Comment by mandrigin** [comment]
@ovx hey! thanks for your contribution.
sorry for making stuff a bit confusing in the issue description 😅 

can you, please, update your PR so it contains only this: `web3_clientversion` returns the current version and a revision (git commit sha) along another information.
example output is: `"result": "Status/v0.9.3-729b57db/darwin/go1.4.1"`

we reconsidered the approach, and I forgot to update the issue accordingly, sorry for that :(

**Comment by ovx** [comment]
@mandrigin ok, no problem. If I understand it correctly, new RPC handler `web3_clientversion` needs to be registered here https://github.com/status-im/status-go/blob/develop/geth/api/backend.go#L262 (`registerHandlers `)

**Comment by adambabik** [comment]
@ovx `web3_clientversion` is already built-in so no need to register it. I believe what we need to do is to make sure the version passed here https://github.com/status-im/status-go/blob/02309e81e98bdda0317ed35b4c94a234d301ab90/geth/params/config.go#L319 (defined here https://github.com/status-im/status-go/blob/02309e81e98bdda0317ed35b4c94a234d301ab90/geth/params/version.go#L22) has the proper value. 

Git SHA is passed with [`BUILD_FLAGS`](https://github.com/status-im/status-go/blob/02309e81e98bdda0317ed35b4c94a234d301ab90/Makefile#L15), however, it overrides different vars. You need to pass different build tags to override `VersionMeta` in `geth/params` package. Don't care about `VersionMajor` etc. for now, just let's make sure `VersionMeta` has Git SHA value.

How to test it? First, build the binary: `make statusgo`, then run it: `./build/bin/statusd -http` and finally in a different terminal use curl:
```
curl -v -X POST -H 'Content-Type: application/json' --data '{"jsonrpc":"2.0","method":"web3_clientVersion", "params": [], "id":1}' http://127.0.0.1:8545
```

**Comment by ovx** [comment]
Fixed.

`statusd --version` output:

```
StatusIM
Version: 0.9.9-07298451
Git Commit: 07298451
...
    "Version": "0.9.9-07298451",
```

RPC output: 

```
{"jsonrpc":"2.0","id":1,"result":"StatusIM/v0.9.9-07298451/darwin-amd64/go1.10"}
```

**Review by mandrigin — APPROVED** [review]
Looks good! @adambabik?

**Review by adambabik — APPROVED** [review]
LGTM! Very nice and short solution!

**Pull request merged**